### PR TITLE
migrating webapp/channels/src/tests files to typescript

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
This migrates some extra files to typescript.

#### Ticket Link
Not yet

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/tests/helpers/markdown_hotkey_helpers.ts [hunk 1]
```diff
@@ -82,8 +95,9 @@ function makeLinkHotKeyEvent(input, start, end) {
  * @param  {function} getValue - single parameter for the React Component instance
  * NOTE: runs Jest tests
  */
-export function testComponentForMarkdownHotkeys(generateInstance, initRefs, find, getValue, intlInjected = true, targetId) {
-    const shallowRender = intlInjected ? shallowWithIntl : shallow;
+export function testComponentForMarkdownHotkeys(generateInstance: (input: string) => JSX.Element, initRefs: (instance: ShallowWrapper, setSelectionRange: (start: number, end: number) => void) => void, find: (instance: ShallowWrapper) => JSX.Element, getValue: (instance: React.Component<any, any>|ShallowWrapper) => string, intlInjected = true, targetId: string) {
```
Comment (larkox) [inline webapp/channels/src/tests/helpers/markdown_hotkey_helpers.ts:98]: Just for legibility, I would separate this in several lines
```suggestion
export function testComponentForMarkdownHotkeys(
    generateInstance: (input: string) => JSX.Element,
    initRefs: (instance: ShallowWrapper, setSelectionRange: (start: number, end: number) => void) => void,
    find: (instance: ShallowWrapper) => JSX.Element,
    getValue: (instance: React.Component<any, any>|ShallowWrapper) => string,
    intlInjected = true,
    targetId: string,
) {
```

**Review by larkox — APPROVED** [review]
LGTM, just one minor improvement.

**Comment by jespino** [comment]
@lindy65 This is mainly tests updates son there is much affecting the application code. Anyway we change just a tiny bit of the code in the post editor, channel header editor, post creation, thread post creation and forward post functionalities. So I guess that is what we need to QA.

**Comment by lindy65** [comment]
Thanks @jespino - I've added the `qa-wanted` and `hacktoberfest` labels but if no community QA picks it up to test within the next 48 hours, I'll test it 👍

**Comment by lindy65** [comment]
QA contributor, Jelena, will be testing this - waiting for her to comment on the PR so I can assign her

**Comment by jgilliam17** [comment]
Thanks @lindy65 - please assign it to me for testing

**Comment by jgilliam17** [comment]
I tested editing posts, editing channel header, creating posts and threads and forwarding posts.  No issues, looks good to merge.

**Review by lindy65 — APPROVED** [review]
Thanks @jespino - looks good 👍 

Thanks for QA'ing @jgilliam17 🙌

**Pull request merged**